src/main/java/edu/illinois/cs/dt/tools/detection/SmartShuffler.java

```diff
@@ -33,27 +38,38 @@ public SmartShuffler(final List<String> tests) {
             final String className = TestShuffler.className(test);
 
             if (!classToMethods.containsKey(className)) {
-                classToMethods.put(className, new RandomList<>());
+                classToMethods.put(className, new ArrayList<>());
             }
 
             classToMethods.get(className).add(test);
             methodToClass.put(test, className);
         }
+
+        // Set up Random instance using passed in seed, if available
+        int seed = 42;
+        try {
+            seed = Integer.parseInt(Configuration.config().getProperty("dt.seed", "42"));
+        } catch (NumberFormatException nfe) {
+            TestPluginUtil.project.info("dt.seed needs to be an integer, using default seed " + seed);
+        }
+        this.random = new Random(seed);
     }
 
-    private RandomList<String> testSiblings(final String testName) {
+    private List<String> testSiblings(final String testName) {
         return classToMethods.get(methodToClass.get(testName));
     }
 
     @SafeVarargs
     private final void addTestMethods(final List<String> order, final Optional<String>... excluding) {
         // Add all classes other than the last one
-        for (final RandomList<String> methods : classToMethods.values()) {
+        for (final List<String> methods : classToMethods.values()) {
             final boolean foundExcludedTest =
                     Arrays.stream(excluding).anyMatch(t -> t.isPresent() && methods.contains(t.get()));
 
             if (!foundExcludedTest) {
-                order.addAll(methods.shuffled());
+                List<String> methodsShuffled = new ArrayList<>(methods);
+                Collections.shuffle(methods, random);
```

Comment:
Don't think we ever really use the SmartShuffler but should we deterministically sort the methods (e.g., alphabetically like these changes do for TestShuffler) before shuffling the methods? The same would apply to the shuffling of siblings in this class.

Reply:
Probably good to sort. I think it mainly ensures that even if the starting list of original tests has a different order they would still result in the same kind of deterministic random orderings with the same seed.

Reply:
One solution for future changes to reduce the overhead can be to cache the sorted list. E.g., we maintain a map where the keys are a set of test names and the values are the sorted list, then future calls to get a random order with the same set of tests would not need to sort again. 

Anyways, the overhead created by these changes are fine with me for now. Merging now...